meson.build

```diff
@@ -0,0 +1,1629 @@
+link_flags = []
+
+test_flags = ['-Werror=implicit-function-declaration',
+              '-Wno-error=deprecated-declarations',
+              '-Wno-error=unused-function',
+              '-Wempty-body',
+              '-Wdisabled-optimization',
+              '-Wstrict-prototypes',
+              '-Wno-format-zero-length',
+              '-Werror=format-security',
+              '-Wno-redundant-decls',
+              '-Wvla',
+              '-Wno-format-truncation',
+              '-Wimplicit-fallthrough',
+              '-fno-math-errno']
+
+flags += cc.get_supported_arguments(test_flags)
+
+if cc.get_id() == 'gcc'
+    gcc_flags = ['-Wundef', '-Wmissing-prototypes', '-Wshadow',
+                 '-Wno-switch', '-Wparentheses', '-Wpointer-arith',
+                 '-Wno-pointer-sign',
+                 # GCC bug 66425
+                 '-Wno-unused-result']
+    flags += gcc_flags
+endif
+
+if cc.get_id() == 'clang'
+    clang_flags = ['-Wno-logical-op-parentheses', '-fcolor-diagnostics',
+                   '-Wno-tautological-compare',
+                   '-Wno-tautological-constant-out-of-range-compare']
+    flags += clang_flags
+endif
+
+vector = {
+    'name': 'vector',
+    'statement': run_command('cat', 'waftools/fragments/vector.c').stdout(),
+    'use': false,
+}
+vector += {'use': cc.compiles(vector['statement'], name: 'vector check')}
+
+win32 = host_machine.system() == 'cygwin' or host_machine.system() == 'windows'
+mingw = win32 # assume true for now but this is probably wrong
+posix = false
+if not win32
+    posix = true
+endif
+
+mswin_flags = ['-D_WIN32_WINNT=0x0602', '-DUNICODE', '-DCOBJMACROS',
+               '-DINITGUID', '-U__STRICT_ANSI__']
+
+mswin_linkflags = ['-Wl,--major-os-version=6,--minor-os-version=0',
+                   '-Wl,--major-subsystem-version=6,--minor-subsystem-version=0']
+
+if host_machine.system() == 'windows'
+    flags += [mswin_flags, '-D__USE_MINGW_ANSI_STDIO=1']
+    link_flags += [mswin_linkflags, '-mwindows']
+endif
+
+if host_machine.system() == 'cygwin'
+    flags += [mswin_flags, '-mwin32']
+    link_flags += mswin_linkflags
+endif
+
+noexecstack = false
+if cc.has_link_argument('-Wl,-z,noexecstack')
+    link_flags += '-Wl,-z,noexecstack'
+    noexecstack = true
+endif
+
+if cc.has_link_argument('-Wl,--nxcompat,--no-seh,--dynamicbase')
+    link_flags += '-Wl,--nxcompat,--no-seh,--dynamicbase'
+    noexecstack = true
+endif
+
+if get_option('build-date').disabled()
+    flags += '-DNO_BUILD_TIMESTAMPS'
+endif
+
+if get_option('pthread-debug').enabled()
+    if cc.has_argument('-DMP_PTHREAD_DEBUG')
+        flags += '-DMP_PTHREAD_DEBUG'
+    endif
+endif
+
+libdl = {
+    'name': 'libdl',
+    'deps': cc.find_library('libdl', required: false),
+    'use' : false,
+}
+if libdl['deps'].found()
+    libdl_fragment = run_command('cat', 'waftools/fragments/libdl.c').stdout()
+    libdl += {'use': cc.compiles(libdl_fragment, dependencies: libdl['deps'], name: 'libdl check')}
```

Comment:
more run_command + cat when files() can be directly checked. :)

Also I don't entirely get the point of this. If you do find libdl, do you really need to add all-new code not present in waf, to check whether libdl works?